packages/server-wallet/src/chain-service/types.ts

```diff
@@ -45,6 +50,7 @@ interface ChainModifierInterface {
   concludeAndWithdraw(
     finalizationProof: SignedState[]
   ): Promise<providers.TransactionResponse | void>;
+  pushOutcomeAndWithdraw(state: State): Promise<providers.TransactionResponse>;
```

Comment:
It's worth discussing whether or not there should just be a single API on the `ChainService` for both of these cases, something equivalent to the wallet saying "here is everything I have, please get the money out of this channel", and then letting the `ChainService` decide what to do when it gets called. The benefit would be that the `Wallet` would not need to be as concerned about what is on-chain and can rely on the `ChainService` to figure out any of those details.

Reply:
That's a good point! https://github.com/statechannels/statechannels/issues/3048